packages/core/src/cart/cart-request-sender.ts

```diff
@@ -19,4 +47,136 @@ export default class CartRequestSender {
 
         return this._requestSender.post(url, { body, headers, timeout });
     }
+
+    loadCardEntity(cartId: string, options: LoadCartRequestOptions): Promise<Response<Cart>> {
+        const url = `/graphql`;
+
+        const graphQLQuery = `
+        query {
+            site {
+                cart(entityId: "${cartId}") {
+                  currencyCode
+                  entityId
+                  id
+                  isTaxIncluded
+                  discounts {
+                    entityId
+                    discountedAmount {
+                      currencyCode
+                      value
+                    }
+                  }
+                  discountedAmount {
+                    currencyCode
+                    value
+                  }
+                  baseAmount {
+                    currencyCode
+                    value
+                  }
+                  amount {
+                    currencyCode
+                    value
+                  }
+                  lineItems {
+                    physicalItems {
+                      brand
+                      couponAmount {
+                        value
+                      }
+                      discountedAmount {
+                        value
+                      }
+                      discounts {
+                        discountedAmount {
+                          value
+                        }
+                        entityId
+                      }
+                      extendedListPrice {
+                        value
+                      }
+                      extendedSalePrice {
+                        value
+                      }
+                      giftWrapping {
+                        amount {
+                          value
+                        }
+                        message
+                        name
+                      }
+                      isShippingRequired
+                      isTaxable
+                      listPrice {
+                        value
+                      }
+                      name
+                      originalPrice {
+                        value
+                      }
+                      entityId
+                      quantity
+                      salePrice {
+                        value
+                      }
+                      sku
+                      url
+                    }
+                  }
+                }
+              }
+        }`;
+
+        const requestOptions: LoadCartRequestOptions = {
+            ...options,
+            headers: {
+                ...options.headers,
+                'Content-Type': 'application/json',
+            },
+            body: {
+                query: graphQLQuery,
+            },
+        };
+
+        return this._requestSender
```

Comment:
It seems to me that we're making a cross-origin Ajax request from headless storefront to BC, which then returns a session token and set as a third party cookie. AFAIK, Safari blocks third party cookies. Could you test if the setup works for Safari?

Another point is, Catalyst creates carts on the server-side. Therefore, the browser does not have BC session cookie required to retrieve carts directly from the browser from BC. Could you also test if the setup works for Catalyst?

I believe the solution to both of these problems is to proxy GQL calls through Catalyst or a server-side middleware for non-Catalyst headless storefronts.

Reply:
> For Safari's case we can avoid blocking of third party cookies by enabling/disabling the appropriate settings.

Yes, but this is not a setting we have control over, as we can't force shoppers to relax their privacy setting. I think this issue is a blocker. If we can't make it work for Safari by default, we need to explore alternative options.

> This point will be mentioned in documentation of headless wallet button.

I don't think documentation is sufficient as the solution won't be compatible with headless Catalyst, which creates carts on the server side.

It seems to me that to solve both of these problems, we shouldn't be calling the GQL API directly in the browser for headless storefronts. Instead, requests need to be routed through a server-side proxy, which in most cases will be the headless storefronts.